model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPAInstallationDao.java

```diff
@@ -98,15 +111,72 @@ public Installation findInstallationForVariantByDeviceToken(String variantID, St
                 .getResultList();
     }
 
+
+
+    @Override
+    public Set<String> findAllDeviceTokenForVariantID(String variantID) {
+        TypedQuery<String> query = createQuery(FIND_ALL_DEVICES_FOR_VARIANT_QUERY, String.class);
+        query.setParameter("variantID", variantID);
+        return new HashSet<String>(query.getResultList());
+    }
+
     @Override
-    public List<String> findAllDeviceTokenForVariantIDByCriteria(String variantID, List<String> categories, List<String> aliases, List<String> deviceTypes) {
+    public ResultsStream.QueryBuilder<String> findAllDeviceTokenForVariantIDByCriteria(String variantID, List<String> categories, List<String> aliases, List<String> deviceTypes, final int maxResults, String lastTokenFromPreviousBatch) {
```

Comment:
no need to do it now, but if we want this on other results, please lets have a JIRA for it